client/src/pages/Register.js

```diff
@@ -72,18 +74,28 @@ class Register extends Component {
       confirmPassword: this.state.confirmPassword
     }
     //make http post request to send name,email and password to server
-    const { data } = await axios.post("/api/users", userData)
-    console.log("new user info:", data);
-    //clear input fields
-    this.setState({
-      userName: " ",
-      userEmail: " ",
-      userPassword: "",
-      confirmPassword: " "
-    })
-
-    //direct user to profile page
-    this.props.history.replace("/profile")
+    // const { data } = await axios.post("/api/users", userData)
+    // console.log("new user info:", data);
+
+    await axios.post("/api/users", userData)
+      .then(response => {
+        console.log("success! Data receieved = ", response.data);
```

Comment:
The response object has a status attribute. I think you want to check it to make sure it's a success code from the server. If it's not successful (e.g 400), you can throw an error and display the appropriate message when you catch it.

Reply:
I've always used the fetch api before, if with axios this is unnecessary, then disregard my comment